csharp/ql/src/Security Features/CWE-639/InsecureDirectObjectReference.qhelp

```diff
@@ -0,0 +1,34 @@
+<!DOCTYPE qhelp PUBLIC
+  "-//Semmle//qhelp//EN"
+  "qhelp.dtd">
+<qhelp>
+<overview>
+<p>When an action method accepts an ID parameter used to control which resource (e.g. a comment, a user profile, etc) 
+is being accessed/modified, checks should me made to ensure that the current user is authorized to access that resource. 
+Otherwise, an attacker could access an arbitrary resource by guessing the ID parameter.</p>
+
+</overview>
+<recommendation>
+<p>
+Ensure that the current user is authorized to access the resource of the provided ID.
+</p>
+
+</recommendation>
+<example>
+<p>In the following example, in the case marked BAD, there is no authorization check, so any user is able to edit any comment.
+In the case marked GOOD, there is a check that the current usr matches the author of the comment.</p>
+<sample src="WebFormsExample.cs" />
+<p>The following example shows a similar case for the ASP.NET Core framework. In the third case, the `Authorize` attribute is used
+to restrict the method to only administrators, which are expected to be able to access arbitrary resources. 
+</p>
```

Comment:
```suggestion
<p>The following example shows a similar scenario for the ASP.NET Core framework. As above, the "BAD" case provides an example with no authorization check, and the first "GOOD" case provides an example with a check that the current user authored the specified comment. Additionally, in the second "GOOD" case, the `Authorize` attribute is used to restrict the method to administrators, who are expected to be able to access arbitrary resources.</p>
```

Suggesting some content for extra clarity around which case is being referenced when.